Code/GraphMol/MolDraw2D/MolDraw2D.h

```diff
@@ -79,12 +91,26 @@ class MolDraw2D {
       const std::map<int, double> *highlight_radii = NULL, int confId = -1);
 
   virtual void drawMolecule(
+      const ROMol &mol, const std::string &legend,
+      const std::vector<int> *highlight_atoms = NULL,
+      const std::map<int, DrawColour> *highlight_map = NULL,
+      const std::map<int, double> *highlight_radii = NULL, int confId = -1);
+
+  virtual void drawMolecule(
```

Comment:
It isn't horrible by any means, but I'm all for option classes for these nasty style options.

In fact I would prefer a virtual base that can answer the questions about the rendering styles ala

{code}
class HighlightOptions {
  virtual DrawColour GetColour(const Atom *) const=0;
  virtual DrawColour GetColour(const Bond *) const=0;
 ...
};
{code}

With a default implementation that handles the intent of the original api. Other implementations could be HighlightSmarts and so on.